### PR TITLE
[wasm] Disable `S.T.RegularExpressions FunctionalTests` tests failing due to linker issues

### PR DESCRIPTION
**Pull request by radical**
- Disabling `FunctionalTests/System.Text.RegularExpressions.Tests` because of a linker crash
- Also, ensure that the HighResourceAOT tests are run for regular
  library tests, and EAT.
Issue: https://github.com/dotnet/runtime/issues/71848

- Disabling some tests for System.Text.Json Serialization due to https://github.com/dotnet/runtime/issues/71838

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

**Comment by lewing** [comment]
cc @agocke

**Comment by radical** [comment]
Many jobs failing due to https://github.com/dotnet/runtime/issues/71981 .

I need to disable one more test for the wasm/aot case, which I will push once the current wasm ones are done.

**Comment by radical** [comment]
Some of the failures are fixed by https://github.com/dotnet/runtime/pull/71976 .

And we seem to have [new tests failing](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-71875-merge-8cb29fbe14114d2694/normal-System.Text.Json.Tests/1/console.4f7c68d6.log?helixlogtype=result) with the same linker error now:
```
[01:14:27] fail: [FAIL] System.Text.Json.Serialization.Tests.PropertyVisibilityTestsDynamic.JsonIgnoreCondition_WhenWritingNull_OnValueType_Fail_EmptyJson(type: typeof(System.Text.Json.Serialization.Tests.PropertyVisibilityTests+ClassWithBadIgnoreAttribute))
[01:14:27] info: Assert.Throws() Failure
[01:14:27] info: Expected: typeof(System.InvalidOperationException)
[01:14:27] info: Actual:   typeof(System.NotSupportedException): Metadata for type 'System.Text.Json.Serialization.Tests.PropertyVisibilityTests+ClassWithBadIgnoreAttribute' was not provided to the serializer. The serializer method used does not support reflection-based creation of serialization-related type metadata. If using source generation, ensure that all root types passed to the serializer have been indicated with 'JsonSerializableAttribute', along with any types that might be serialized polymorphically.
[01:14:27] info: ---- System.NotSupportedException : Metadata for type 'System.Text.Json.Serialization.Tests.PropertyVisibilityTests+ClassWithBadIgnoreAttribute' was not provided to the serializer. The serializer method used does not support reflection-based creation of serialization-related type metadata. If using source generation, ensure that all root types passed to the serializer have been indicated with 'JsonSerializableAttribute', along with any types that might be serialized polymorphically.
[01:14:27] info:    at System.Text.Json.Serialization.Tests.JsonSerializerWrapper.StringSerializerWrapper.SerializeWrapper[Object](Object value, JsonSerializerOptions options)
[01:14:27] info:    at System.Text.Json.Serialization.Tests.PropertyVisibilityTests.<>c__DisplayClass172_0.<<JsonIgnoreCondition_WhenWritingNull_OnValueType_Fail_EmptyJson>b__1>d.MoveNext()
[01:14:27] info: --- End of stack trace from previous location ---
[01:14:27] info:    at System.Text.Json.Serialization.Tests.JsonSerializerWrapper.StringSerializerWrapper.SerializeWrapper[Object](Object value, JsonSerializerOptions options)
[01:14:27] info: ----- Inner Stack Trace -----
[01:14:27] info:    at System.Text.Json.Serialization.Tests.JsonSerializerWrapper.StringSerializerWrapper.SerializeWrapper[Object](Object value, JsonSerializerOptions options)
[01:14:27] info:    at System.Text.Json.Serialization.Tests.PropertyVisibilityTests.<>c__DisplayClass172_0.<<JsonIgnoreCondition_WhenWritingNull_OnValueType_Fail_EmptyJson>b__1>d.MoveNext()
[01:14:27] info: --- End of stack trace from previous location ---
[01:14:27] info:    at System.Text.Json.Serialization.Tests.JsonSerializerWrapper.StringSerializerWrapper.SerializeWrapper[Object](Object value, JsonSerializerOptions options)
```

**Comment by radical** [comment]
The coreclr failure is https://github.com/dotnet/runtime/issues/71995 .
The wasm/debugger failure is intermittent.

File: src/libraries/System.Text.Json/tests/Common/PropertyVisibilityTests.cs [hunk 1]
```diff
@@ -1178,7 +1191,7 @@ public async Task JsonIgnoreAttribute_UnsupportedCollection()
 
             var dictionary = new Dictionary<object, object>();
             // Uri is an unsupported dictionary key.
-            dictionary.Add(new Uri("http://foo"), "bar");
+            dictionary.Add(new Uri("http://typeof(PlatformDetection), nameof(PlatformDetection.IsBrowser), nameof(PlatformDetection.IsMonoAOT))]"), "bar");
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/Common/PropertyVisibilityTests.cs:1194]: This looks like an accidental change, could you revert please? cc @krwq

Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/PropertyVisibilityTests.cs:1194]: I'm actually surprised that this doesn't break other platforms.

**Pull request merged**